proposal.rst

```diff
@@ -168,27 +210,83 @@ Project Details
 
 **Describe any existing work and concepts on which your project is based.**
 
-FILLME
+My work will be based on the GSoC 2017 Project `Clickstream Mining for 
+Learning to Rank training data <https://trac.xapian.org/wiki/GSoC2017/LetorClickstream>`_. 
+
+**Phase 1: Implementing Click Models**
+
+The previous work has implemented the simplified DBN model for clickstream mining. 
+I plan to finish the EM algorithm in order to get a better performance.
+
+
+Besides DBN model, I will implement two more click models, the Dependent click Model 
+(DCM) and the User Browsing Model (UBM). Both two are also highly 
+effective to improve document ranking. 
+
+I will first work on an abstract class for all click models and then try to implement 
+each model according to relevant papers. 
+
+
+
+**Phase 2: Documentation**
+
+The second phase of my work will be a detailed documentation of each model I 
+implemented. It will provide the Omega users with a brief view of click models and
+pros & cons of each model. I will continue to work on the clickmodel.rst in the 
+Omega documentation files.
+
+**Phase 3: Evaluation**
+
+The third phase is about the evaluation. With more and more models applied, I 
+believe it will be necessary to provide an evaluation method for the users. I'm 
+going to use CTR estimation for the evaluation of the models. For the method 
+is convenient to implement and has a relatively high accuracy. Also the click logs 
+is easy to acquire due to the previous GSoC project.
+
+
+
 
 **Do you have any preliminary findings or results which suggest that your
 approach is possible and likely to succeed?**
 
-FILLME
+The EM algorithm of **DBN model**: I will implement the EM algorithm for DBN model according to the appendix of 
+`A dynamic bayesian network click model for web search ranking. WWW (2009). 
+<https://dl.acm.org/citation.cfm?id=1526711>`_. 
+
+
+The **DCM model**: I will implement the dependent click model according to the paper `Efficient 
+multiple-click models in web search. WSDM (2009). 
+<https://dl.acm.org/citation.cfm?id=1498818>`_.
+
+The **UBM model**: the user browsing model is from paper `A user browsing model to predict search engine click data from past observations. SIGIR (2008).  <https://dl.acm.org/citation.cfm?id=1390392>`_.
+
 
 **What other approaches to have your considered, and why did you reject those in
 favour of your chosen approach?**
 
-FILLME
+The Federated Click Model (FCM) and Vertical Click Model (VCM)are not appropiate 
+for Xapian because of the SERP (search engine return pages) of Xapian do not 
+provide with vertical contents.
+
+I have also considered about the end to end model.However, I can not make sure to
+develop a clear and fairly smooth process all the way through from logging clicks 
+to turning the logs into relevance judgements to training the model to getting 
+queries reranked. So I didn't continue to work on this idea.
```

Comment:
What would such a model involve? An end-to-end _use case_ of Omega is an important part of the project.